pkgs/development/tools/godot/4/default.nix

```diff
@@ -59,7 +49,17 @@ stdenv.mkDerivation rec {
     owner = "godotengine";
     repo = "godot";
     rev = version;
-    hash = "sha256-g9+CV3HsiJqiSJpZvK0N7BqKzp2Pvi6otjRLsFdmWGk=";
+    hash = "sha256-zSysJe00D0/Zv+oHnHpxQVusorz7msterihx6DDc9MQ=";
+
+    # Required for the commit hash to be included in the version number.
+    #
+    # `methods.py` reads the commit hash from `.git/HEAD` and manually follows
+    # refs. It would suffice to write the actual version hash to `.git/HEAD`,
+    # maybe using `git get-tar-commit-id` on the downloaded tar file.
+    #
+    # See also 'hash' in
+    # https://docs.godotengine.org/en/stable/classes/class_engine.html#class-engine-method-get-version-info
+    leaveDotGit = true;
```

Comment:
If possible you should try to avoid `leaveDotGit = true;` as much as possible since it isn't really reproducible and instead use search and replace to write the version number at the correct position or write a file with the needed content at .git/HEAD as you mentioned in the comment above.

Reply:
The reason I did not do that because I couldn't figure out how to access the tar file during building.

Reply:
You mean the github release tar? As far as I understood it you need the commit hash and the version number which are both attributes already defined in the `.nix` file, can you not re-use those? Sorry if I'm not getting something.

Reply:
I was trying to avoid downloading the tar and use the one the source code is extracted from at the beginning of the build process. I realize now that this probably doesn't matter since the tar should be in the store anyways once it's been downloaded once, so I can just get it again.

Reply:
instead of using `leaveDotGit = true`, I'd suggest patching `methods.py`

Reply:
> You mean the github release tar? As far as I understood it you need the commit hash and the version number which are both attributes already defined in the `.nix` file, can you not re-use those? Sorry if I'm not getting something.

Where is the commit hash defined in the .nix file? Getting the commit hash is the reason I'm trying to access the tar in the first place.

Reply:
it (probably) isn't, but you can just define it. instead of setting `rev = version`, just use the actual commit hash, instead of the tag.

Reply:
or set an extra attribute containing the commit hash.

Reply:
Okay, I'll do the former so it doesn't accidentally get overlooked during updates.